library/datasort.c

```diff
@@ -1687,12 +1687,14 @@ int eblob_generate_sorted_data(struct datasort_cfg *dcfg)
 		goto err_unlock_bctl;
 	}
 
+	pthread_rwlock_wrlock(&dcfg->b->iteration_lock);
```

Comment:
Here we have following order of locks: `b->lock` -> `bctls->lock` -> `b->iteration_lock`.
In iterator we have another order of locks: `b->iteration_lock` -> `bctl->lock`.

This may lead to deadlock.

Reply:
fixed